Fix compilation errors with btrfs-progs-4.5

devimc opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/docker/docker/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

btrfs-progs-4.5 introduces device delete by devid
for this reason btrfs_ioctl_vol_args_v2's name was encapsulated
in a union

this patch is for setting btrfs_ioctl_vol_args_v2's name
using a C function in order to preserve compatibility
with all btrfs-progs versions

Signed-off-by: Julio Montes imc.coder@gmail.com

thaJeztah commented:
@devimc looks like this needs a gofmt; https://jenkins.dockerproject.org/job/Docker-PRs-experimental/17233/console

```
21:40:08 These files are not properly gofmt'd:
21:40:08  - daemon/graphdriver/btrfs/btrfs.go
21:40:08 
21:40:08 Please reformat the above files using "gofmt -s -w" and commit the result.
```

devimc commented:
ohh my bad, sorry

devimc committed:
Fix compilation errors with btrfs-progs-4.5

btrfs-progs-4.5 introduces device delete by devid
for this reason btrfs_ioctl_vol_args_v2's name was encapsulated
in a union

this patch is for setting btrfs_ioctl_vol_args_v2's name
using a C function in order to preserve compatibility
with all btrfs-progs versions

Signed-off-by: Julio Montes <imc.coder@gmail.com>
